From af0d75ad8695b57a83157b4ed9b30cc1e447f176 Mon Sep 17 00:00:00 2001
From: Amelie Delaunay <amelie.delaunay@foss.st.com>
Date: Fri, 31 Jan 2025 13:11:48 +0100
Subject: [PATCH 1496/1695] dts: stm32: add OEM ADAC ROTPK hash for stm32mp21

ADAC (Authenticated Debug Access Control) requires a ROTPK (Root Of Trust
Public Key) to verify certificate and token signatures during
authentication process.
ECDSA algorithm is used in conjunction with NIST P-256 curve.
Hash (256 bits) of this ROTPK is stored in OTP238-245.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/optee_os.git 0002b381567afdbdad73d7d59d2d92d32fbab5d4 dts: stm32: add OEM ADAC ROTPK hash for stm32mp21]
Signed-off-by: Amelie Delaunay <amelie.delaunay@foss.st.com>
Change-Id: I6eed9f7069ebf80654bae6c809919dec5408a567
Reviewed-on: https://gerrit.st.com/c/mpu/oe/optee/optee_os/+/433594
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
Domain-Review: Yann GAUTIER <yann.gautier@foss.st.com>
Reviewed-by: Thomas BOURGOIN <thomas.bourgoin@foss.st.com>
---
 core/arch/arm/dts/stm32mp211.dtsi | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/core/arch/arm/dts/stm32mp211.dtsi b/core/arch/arm/dts/stm32mp211.dtsi
index 0cb604972..bf354fc06 100644
--- a/core/arch/arm/dts/stm32mp211.dtsi
+++ b/core/arch/arm/dts/stm32mp211.dtsi
@@ -653,6 +653,12 @@
 				st,non-secure-otp-provisioning;
 			};
 
+			oem_adac_rotpkh: oem_adac_rotpkh@3b8 {
+				/* 0x3b8 -> otp238 */
+				reg = <0x3b8 0x20>;
+				st,non-secure-otp-provisioning;
+			};
+
 			oem_fip_enc_key: otp260@410 {
 				reg = <0x410 0x20>;
 				st,non-secure-otp-provisioning;
-- 
2.47.2

